databricks/labs/doc_qa/model_generators/model_generator.py

```diff
@@ -525,3 +518,93 @@ def _generate(
             is_successful=True,
             error_msg=None,
         )
+
+
+# The model generator based on the openAI format implemented by the vLLM framework:
+# Reference: https://vllm.readthedocs.io/en/latest/getting_started/quickstart.html#openai-compatible-server
+class vLllmOpenAICompletionFormatModelGenerator(BaseModelGenerator):
+    def __init__(
+        self,
+        model_name,
+        url: str,
+        pat_token: str,
+        format_prompt_func: callable,
+        prompt_formatter: PromptTemplate,
+        batch_size: int = 1,
+        concurrency: int = 64,
+    ) -> None:
+        """
+        Args:
+            prompt_formatter (PromptTemplate): the prompt format to format the input dataframe into prompts row by row according to the column names
+            model_name (str): the model name
+            batch_size (int, optional): Batch size that will be used to run tasks. Defaults to 1, which means it's sequential.
+        """
+        super().__init__(prompt_formatter, batch_size, concurrency)
+        # check the batch_size can only be 1 for this model generator
+        if batch_size != 1:
+            raise ValueError(
+                f"batch_size {batch_size} is not supported for {self.__class__.__name__}, only 1 is supported"
+            )
+        self._model_name = model_name
+        self._url = url
+        self._pat_token = pat_token
+        self._format_prompt_func = format_prompt_func
+
+    def _generate(
+        self, prompts: list, temperature: float, max_tokens=256, system_prompt=None
+    ) -> BatchGenerateResult:
+        if temperature == 0.0:
+            top_p = 1
+        else:
+            top_p = 0.95
+
+        all_formatted_prompts = [
+            self._format_prompt_func(message=message, system_prompt_opt=system_prompt)
+            for message in prompts
+        ]
+
+        import requests
```

Comment:
Can you add "isort" to CI for this repo? :) it'll sort the imports to be at the top of the file

Reply:
Oh, this is intentional, as the file grow I might move the module outside, and don't want to introduce global dependency until people use the module, thus trying to constraint the import only inside the module.